client/dashboard/sites/hooks/use-can-transfer-site.ts

```diff
@@ -0,0 +1,15 @@
+import { useAuth } from '../../app/auth';
+import type { Site } from '../../data/types';
+
+export function useCanTransferSite( { site }: { site?: Site } ) {
+	const { user } = useAuth();
```

Comment:
cc: @taipeicoder 

This can't be backported to v1 easily. I'd suggest moving this as far outside as possible for now; we can for example move this `useAuth()` call to the `<SettingsTransferSite/>` component, make `user` as param to this function, then

```js
const { user } = useAuth();
...

if ( ! canTransferSite( { user, site } ) ) {
...
```

Reply:
After that then we can figure out how to pass the `user` to the component from v1 later 😄

Reply:
We don't need to worry about this as the check will be moved to the endpoint 😄

Reply:
It's a small thing and I don't mind the current hook or utility... But 

 - I'm not sure the current hook brings any value compared to just inline the check in the component
 - If we are to start adding hooks and utils like this (which we started doing), let's be consistent and have a unique well thought strategy.

(For clarity this is a non-blocking comment)